ui/raidboss/data/06-ew/raid/p12s.ts

```diff
@@ -3020,18 +3037,148 @@ const triggerSet: TriggerSet<Data> = {
         return { alertText: output.darkTowerSwitch!() };
       },
     },
+    {
+      id: 'P12S Summon Darkness Preposition',
+      type: 'StartsUsing',
+      netRegex: { id: '832F', source: 'Pallas Athena', capture: false },
+      condition: (data) => data.seenSecondTethers === false,
+      alertText: (_data, _matches, output) => output.stackForTethers!(),
+      outputStrings: {
+        stackForTethers: {
+          en: 'Stack for Tethers',
+        },
+      },
+    },
+    {
+      id: 'P12S Ultima Ray 1',
+      type: 'StartsUsing',
+      netRegex: { id: '8330', source: 'Hemitheos' },
+      condition: (data) => data.phase === 'gaiaochos1',
+      infoText: (data, matches, output) => {
+        data.darknessClones.push(matches);
+        if (data.darknessClones.length !== 3)
+          return;
+
+        // during 'UAV' phase, the center of the circular arena is [100, 90]
+        const uavCenterX = 100;
+        const uavCenterY = 90;
+
+        const unsafeMap: Partial<Record<DirectionOutput8, DirectionOutput8>> = {
+          dirN: 'dirS',
+          dirNE: 'dirSW',
+          dirE: 'dirW',
+          dirSE: 'dirNW',
+          dirS: 'dirN',
+          dirSW: 'dirNE',
+          dirW: 'dirE',
+          dirNW: 'dirSE',
+        };
+        let safeDirs = Object.keys(unsafeMap) as DirectionOutput8[];
```

Comment:
```suggestion
        let safeDirs = Directions.output8Dir;
```

(In general, I try to avoid using `as` as much as possible, because it's defeating the type system and thus could be wrong.)